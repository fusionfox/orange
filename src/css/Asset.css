.Asset {
  text-align: center;
}

.Asset__Title {
  margin-top: 50px;
}

.Asset__Description {
  padding: 0 5%;
}

.Asset__Image {
  width: 80%;
}
