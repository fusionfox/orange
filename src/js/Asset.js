import React, { Component } from 'react';
import axios from 'axios'
import './../css/Asset.css';
import Title from './components/Title'
import ReactImageFallback from "react-image-fallback";
import nasaImageDataAdapter from './utils/nasaImageDataAdapter'
import blankImage from '../images/blank.png'

axios.defaults.baseURL = 'https://images-api.nasa.gov'

class Asset extends Component {
  constructor() {
    super()
    this.state = {
      image: {}
    }
  }

  componentDidMount() {
    axios.get(`/search?nasa_id=${this.props.match.params.id}`)
      .then((response) => response.data.collection.items.map(nasaImageDataAdapter))
      .then((data) => this.setState({image: data[0]}))
      .catch((error) => {
        console.log(error);
      })
  }

  render() {
    return (
      <div className="Asset">
        <Title className="Asset__Title" text={this.state.image.title} />
        <p className="Asset__Description">{this.state.image.description}</p>
        <ReactImageFallback className="Asset__Image" src={this.state.image.large} fallbackImage={this.state.image.thumbnail||blankImage} alt={this.state.image.title} />
      </div>
    );
  }
}

export default Asset;
