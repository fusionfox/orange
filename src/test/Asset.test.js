import React from 'react'
import renderer from 'react-test-renderer'
import Asset from '../js/Asset'

test('Asset rendered correctly', () => {
  const match = {
    params: {
      id: 'PIA00231'
    }
  }

  const component = renderer.create(
    <Asset match={match} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
