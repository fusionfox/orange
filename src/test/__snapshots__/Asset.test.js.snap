// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset rendered correctly 1`] = `
<div
  className="Asset"
>
  <h1
    className="Asset__Title Title"
  />
  <p
    className="Asset__Description"
  />
  <img
    alt={undefined}
    className="Asset__Image"
    src="../images/blank.png"
  />
</div>
`;
